src/main/java/com/ibm/stocator/fs/swift/auth/JossAccount.java

```diff
@@ -109,4 +119,27 @@ public Account getAccount() {
     }
     return mAccount;
   }
+
+  /**
+   * Creates a thread safe HttpClient to use for requests
+   */
+  private HttpClient initHttpClient() {
+    PoolingHttpClientConnectionManager manager = new PoolingHttpClientConnectionManager();
+    manager.setDefaultMaxPerRoute(25);
+    manager.setMaxTotal(50);
+    SocketConfig socketConfig = SocketConfig.custom()
+                                            .setSoKeepAlive(false)
+                                            .setSoTimeout(60000)
```

Comment:
@djalova  One thing  that miss here is the retry mechanism. It was in DefaultHTTP.. but here it's miss. Therefore in case of timeout there is no retry. I saw posts that it can be added to the custom().
Here is some info on retry 
http://stackoverflow.com/questions/23054289/httpclient-executes-requests-multiple-time-if-request-timed-out